imagetest/test_suites/security/image_security_test.go

```diff
@@ -0,0 +1,336 @@
+	}
+	if !strings.Contains(image, "debian") && !strings.Contains(image, "ubuntu") && !strings.Contains(image, "sles") {
+		if err := verifyServiceEnabled(image); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// Check that the security packages are marked for automatic update
+	// https://wiki.debian.org/UnattendedUpgrades
+	// https://help.ubuntu.com/community/AutomaticSecurityUpdates
+	if err := verifySecurityUpgrade(image); err != nil {
+		t.Fatal(err)
+	}
+	if err := verifyAutomaticUpdate(image); err != nil {
+		t.Fatal(err)
+	}
+}
+
+// TestPasswordSecurity Ensure that the system enforces strong passwords and correct lockouts.
+func TestPasswordSecurity(t *testing.T) {
+	if err := verifySSHConfig(); err != nil {
+		t.Fatal(err)
+	}
+
+	// Root password/login is disabled.
+	if err := verifyPassword(); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func verifyPassword() error {
+	image, err := utils.GetMetadata("image")
+	if err != nil {
+		return fmt.Errorf("couldn't get image from metadata")
+	}
+	fileBytes, err := ioutil.ReadFile("/etc/passwd")
+	if err != nil {
+		return err
+	}
+	users := strings.Split(string(fileBytes), "\n")
+	for _, user := range users {
+		// ignore empty user
+		if len(user) == 0 {
+			continue
+		}
+		passwdItems := strings.Split(user, ":")
+		loginname := passwdItems[0]
+		passwd := passwdItems[1]
+		uid := passwdItems[2]
+		shell := passwdItems[6]
+		// A passwd entry of length 1 is a disabled password.
+		if len(passwd) != 1 {
+			return fmt.Errorf("login for %s does not look disabled", loginname)
+		}
+
+		uidValue, err := strconv.Atoi(uid)
+		if err != nil {
+			return err
+		}
+		// Don't check user account and root account
+		if uidValue >= maxUID || uidValue == 0 {
+			continue
+		}
+		if targetShell, found := specialAccountShells[loginname]; found {
+			if targetShell != shell {
+				return fmt.Errorf("account %s has wrong login shell %s", loginname, shell)
+			}
+		} else {
+			// SUSE has bin user with login access
+			if !strings.Contains(image, "sles") && !strings.Contains(shell, "false") && !strings.Contains(shell, "nologin") {
+				return fmt.Errorf("account %s has the login shell %s", loginname, shell)
+			}
+		}
+	}
+	return nil
+}
+
+func verifySSHConfig() error {
+	fileBytes, err := ioutil.ReadFile("/etc/ssh/sshd_config")
+	if err != nil {
+		return err
+	}
+	sshdConfig := string(fileBytes)
+	if !strings.Contains(sshdConfig, "PasswordAuthentication no") {
+		return fmt.Errorf("\"PasswordAuthentication\" was not set to \"no\"")
+	}
+
+	noRootSSH := strings.Contains(sshdConfig, "PermitRootLogin no")
+	noRootPasswordSSH := strings.Contains(sshdConfig, "PermitRootLogin without-password")
+	if !noRootSSH && !noRootPasswordSSH {
+		return fmt.Errorf("\"PermitRootLogin\" was not set to \"no\" or \"without-password\"")
+	}
+	return nil
+}
+
+func verifySecurityUpgrade(image string) error {
+	var expectedBlock, expectedLine string
+	switch {
+	case isRhelbasedLinux(image):
```

Comment:
if it is invalid to call this function for non-debian, then you should only return err, not hide a nil here. move the image switch to the main test function